AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Dependency/AuthenticationProviderAdapter+SignIn.swift

```diff
@@ -260,9 +260,25 @@ extension AuthenticationProviderAdapter {
                 break
             }
         }
+        if self.isErrorCausedByBadRequest(error) {
```

Comment:
this will be a breaking change right? wondering whether this should go as minor version?

Reply:
Perhaps.  It essentially changes the error that is returned from this:
```
▿ AuthError: Unexpected error occurred with message: An unknown error occurred
Recovery suggestion: This should not happen. There is a possibility that there is a bug if this error persists. Please take a look at https://github.com/aws-amplify/amplify-ios/issues to see if there are any existing issues that match your scenario, and file an issue with the details of the bug if there isn't. Issue encountered at:
file: /Users/ametec/Developer/amplify-ios/Amplify/Categories/Auth/Error/AuthError.swift
function: recoverySuggestion
line: 80
Caused by:
Error Domain=com.amazon.cognito.AWSCognitoAuthErrorDomain Code=-3000 "(null)" UserInfo={error=invalid_request: PreAuthentication failed with error This is a custom error message from a Lambda. }
  ▿ unknown: (2 elements)
    - .0: "An unknown error occurred"
    ▿ .1: Optional(Error Domain=com.amazon.cognito.AWSCognitoAuthErrorDomain Code=-3000 "(null)" UserInfo={error=invalid_request: PreAuthentication failed with error This is a custom error message from a Lambda. })
      - some: Error Domain=com.amazon.cognito.AWSCognitoAuthErrorDomain Code=-3000 "(null)" UserInfo={error=invalid_request: PreAuthentication failed with error This is a custom error message from a Lambda. } #0
        - super: NSObject
```
to this:
```
▿ AuthError: invalid_request: PreAuthentication failed with error This is a custom error message from a Lambda.
Recovery suggestion: Retry the webUi signIn
Caused by:
Error Domain=com.amazon.cognito.AWSCognitoAuthErrorDomain Code=-3000 "(null)" UserInfo={error=invalid_request: PreAuthentication failed with error This is a custom error message from a Lambda. }
  ▿ service: (3 elements)
    - .0: "invalid_request: PreAuthentication failed with error This is a custom error message from a Lambda."
    - .1: "Retry the webUi signIn"
    ▿ .2: Optional(Error Domain=com.amazon.cognito.AWSCognitoAuthErrorDomain Code=-3000 "(null)" UserInfo={error=invalid_request: PreAuthentication failed with error This is a custom error message from a Lambda. })
      - some: Error Domain=com.amazon.cognito.AWSCognitoAuthErrorDomain Code=-3000 "(null)" UserInfo={error=invalid_request: PreAuthentication failed with error This is a custom error message from a Lambda. } #0
        - super: NSObject
```
It could be breaking if someone were checking for .unknown errors because it is now returned as a .service error.  With that said, the only useful information returned previously was in the underlying error, which is unchanged.